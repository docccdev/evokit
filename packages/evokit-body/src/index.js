import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
