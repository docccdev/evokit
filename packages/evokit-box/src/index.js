// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
