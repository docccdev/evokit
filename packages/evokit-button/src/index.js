import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
