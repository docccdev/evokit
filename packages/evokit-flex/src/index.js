import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
