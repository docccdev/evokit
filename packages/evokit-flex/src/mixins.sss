@define-mixin media $selector
    $selector
        @mixin-content

    @media (--ek-flex-media-small)
        $(selector)\@small
            @mixin-content

    @media (--ek-flex-media-medium)
        $(selector)\@medium
            @mixin-content

    @media (--ek-flex-media-large)
        $(selector)\@large
            @mixin-content

    @media (--ek-flex-media-wide)
        $(selector)\@wide
            @mixin-content

    @media (--ek-flex-media-huge)
        $(selector)\@huge
            @mixin-content
