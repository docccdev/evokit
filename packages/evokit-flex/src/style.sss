@import './mixins.sss'

@custom-media --ek-flex-media-small only screen and (min-width: 480px)
@custom-media --ek-flex-media-medium only screen and (min-width: 768px)
@custom-media --ek-flex-media-large only screen and (min-width: 960px)
@custom-media --ek-flex-media-wide only screen and (min-width: 1200px)
@custom-media --ek-flex-media-huge only screen and (min-width: 1400px)

.flex
    display: flex
    flex-wrap: wrap
    align-content: flex-start
    align-items: flex-start
    min-width: 0

.flex__item
    display: block
    box-sizing: border-box
    min-width: 0
    min-height: 0

/**
 * Item display:
 *
 *  - flex__item_display_block
 *  - flex__item_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .flex__item_display_$(value)
        display: $value

/**
 * Item align:
 *
 *  - flex__item_align_start
 *  - flex__item_align_end
 *  - flex__item_align_center
 *  - flex__item_align_baseline
 *  - flex__item_align_stretch
 *
 **/

@each $value in (start, end, center, baseline, stretch)
    @mixin media .flex__item_align_$(value)
        @if $value == start
            align-self: flex-start
        @else if $value == end
            align-self: flex-end
        @else
            align-self: $value

/**
 * Item order:
 *
 *  - flex__item_order_*
 *
 **/

@each $value in (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @mixin media .flex__item_order_$(value)
        order: $value

/**
 * Item empty:
 *
 *  - flex__item_empty_hidden
 *
 **/

@mixin media .flex__item_empty_hidden
    &:empty
        display: none

/**
 * Display:
 *
 *  - flex_display_flex
 *  - flex_display_inline-flex
 *  - flex_display_none
 *
 **/

@each $value in (flex, inline-flex, none)
    @mixin media .flex_display_$(value)
        display: $value

/**
 * Content aligment:
 *
 *  - flex_content_*
 *  - flex_content-align_*
 *  - flex_content-justify_*
 *
 **/

@each $value in (start, end, center, between, around, stretch)
    @mixin media .flex_content_$(value)
        @if $value == start
            align-content: flex-start
            justify-content: flex-start
        @else if $value == end
            align-content: flex-end
            justify-content: flex-end
        @else if $value == between
            align-content: space-between
            justify-content: space-between
        @else if $value == around
            align-content: space-around
            justify-content: space-around
        @else
            align-content: $value
            justify-content: $value

    @mixin media .flex_content-align_$(value)
        @if $value == start
            align-content: flex-start
        @else if $value == end
            align-content: flex-end
        @else if $value == between
            align-content: space-between
        @else if $value == around
            align-content: space-around
        @else
            align-content: $value

    @mixin media .flex_content-justify_$(value)
        @if $value == start
            justify-content: flex-start
        @else if $value == end
            justify-content: flex-end
        @else if $value == between
            justify-content: space-between
        @else if $value == around
            justify-content: space-around
        @else
            justify-content: $value

/**
 * Items aligment:
 *
 *  - flex_items_start
 *  - flex_items_end
 *  - flex_items_center
 *  - flex_items_baseline
 *  - flex_items_stretch
 *
 **/

@each $value in (start, end, center, baseline, stretch)
    @mixin media .flex_items_$(value)
        @if $value == start
            align-items: flex-start
        @else if $value == end
            align-items: flex-end
        @else
            align-items: $value

/**
 * Direction:
 *
 *  - flex_direction_row
 *  - flex_direction_row-reverse
 *  - flex_direction_column
 *  - flex_direction_column-reverse
 *
 **/

@each $value in (row, row-reverse, column, column-reverse)
    @mixin media .flex_direction_$(value)
        flex-direction: $value


/**
 * Wrap:
 *
 *  - flex_wrap_nowrap
 *  - flex_wrap_wrap
 *  - flex_wrap_wrap-reverse
 *
 **/

@each $value in (nowrap, wrap, wrap-reverse)
    @mixin media .flex_wrap_$(value)
        flex-wrap: $value

/**
 * Height:
 *
 *  - flex_height_auto
 *  - flex_height_inherit
 *  - flex_height_1-1
 *
 **/

@each $value in (auto, inherit, 1-1)
    @mixin media .flex_height_$(value)
        @if $value == 1-1
            height: 100%
        @else
            height: $value
