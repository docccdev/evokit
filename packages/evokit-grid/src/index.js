import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
