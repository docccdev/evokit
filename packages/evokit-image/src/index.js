import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
