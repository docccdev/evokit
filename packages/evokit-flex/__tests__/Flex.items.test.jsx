import React from 'react';
import { shallow } from 'enzyme';
import { Flex } from '../src';

describe('<Flex flex-items="..." />', () => {
    it('props flex-items', () => {
        const wrapper = shallow(<Flex flex-items='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-flex ek-flex_items_value"></div>');
    });
});
