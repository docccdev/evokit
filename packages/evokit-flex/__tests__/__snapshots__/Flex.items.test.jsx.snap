// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex flex-items="..." /> props flex-items 1`] = `
<div
  className="ek-flex ek-flex_items_value"
/>
`;
